### PR TITLE
Views gets list of lyrics rodney

### PR DESCRIPTION
**Pull request by RJ-0605**
this PR is for the changes made in the api , changes to the view and the url  in the api app

### CODE REVIEW

**Comment by kali-physi-hacker** [comment]
You should link this PR to its issue.

File: api/views.py [hunk 1]
```diff
@@ -1,7 +1,42 @@
...
+
 
 def test(request):
     template = ""
     context = {}
     return render(request=request, template_name=template, context=context)
+
+
+@api_view(["GET"])
+def Get_LyricsList(request): # new
```
Comment (kali-physi-hacker) [inline api/views.py:21]: this is not a good naming convention. functions should always be in small letters. You could just use `lyrics_list` that's fine

File: api/views.py [hunk 2]
```diff
@@ -1,7 +1,42 @@
...
+
+
+@api_view(["GET"])
+def Get_LyricsList(request): # new 
+
+
+    try :
+        # first pick info from model
+        lyrics = Lyric.objects.all()
+    except Lyric.DoesNotExist:
```
Comment (kali-physi-hacker) [inline api/views.py:27]: There was no need for this (I mean the `try` and `except`. Basically, when you do `Lyric.objects.all()`, a queryset is returned, This means when there are no lyrics no exception is raised but rather an empty set (or a set with no elements). So there's no need to wrap in a `try` and `except` block.

File: api/views.py [hunk 3]
```diff
@@ -1,7 +1,42 @@
...
+        return HttpResponse(status=404)
```
Comment (kali-physi-hacker) [inline api/views.py:28]: And I don't know why you're returning an `HttpResponse` here???, even if the above logic, would work, you should not be returning `HttpResponse` but rather `Response` (a djangorestframework modified version for api responses).

File: api/views.py [hunk 4]
```diff
@@ -1,7 +1,42 @@
...
+        
+        if request.method == "GET":
+            # serialize info
+            serializer = LyricSerializer(lyrics, many=True)
+            # return info 
+            return Response (serializer.data)
```
Comment (kali-physi-hacker) [inline api/views.py:34]: The was all you needed. not the above code. Also, to remember that **explicit is better than implicit**. so the last line should have been `return Response(datat=serializer.data, status=status.HTTP_200_OK).

File: api/views.py [hunk 5]
```diff
@@ -1,7 +1,42 @@
...
+
+
+
+    
+    # return info 
+    # check if info can be gotten at once and serialized too
+
```
Comment (kali-physi-hacker) [inline api/views.py:41]: you shouldn't be leaving unneeded comments in PRs

**Review by kali-physi-hacker — CHANGES_REQUESTED** [review]

**Comment by kali-physi-hacker** [comment]
Please make those changes asap. And remember to link the issue to the PR. We are wasting a lot of time over these.

File: api/urls.py [hunk 6]
```diff
@@ -1,5 +1,9 @@
 from django.urls import path
 
-from .views import test
+from .views import test , lyrics_list 
 
-urlpatterns = [path("/test/", test, name="test")]
+urlpatterns = [
+    path("/test/", test, name="test"),\
+    path("/getlyricslist/", lyrics_list, name="lyrics_list")
```
Comment (kali-physi-hacker) [inline api/urls.py:7]: `/lyrics/` should be used instead.

File: api/views.py [hunk 7]
```diff
@@ -1,7 +1,38 @@
...
 def test(request):
     template = ""
     context = {}
     return render(request=request, template_name=template, context=context)
+
+
+@api_view(["GET"])
+def lyrics_list(request): # new 
+
+
```
Comment (kali-physi-hacker) [inline api/views.py:23]: too much spacing.

File: api/views.py [hunk 8]
```diff
@@ -1,7 +1,38 @@
...
+        serializer = LyricSerializer(lyrics, many=True)
+        # return info 
+        return Response ( data=serializer.data , status=status.HTTP_200_OK )
+
+
+
+    
+
+
+
```
Comment (kali-physi-hacker) [inline api/views.py:38]: running black will clear all these whitespace.

**Review by kali-physi-hacker — CHANGES_REQUESTED** [review]

File: api/views.py [hunk 9]
```diff
@@ -1,7 +1,28 @@
...
 
+from rest_framework import status
+from rest_framework.decorators import api_view
+from rest_framework.response import Response
+
+
+from lyrics.models import Lyric
+from lyrics.serializers import LyricSerializer
+
+from django.http import HttpResponse, JsonResponse
```
Comment (kali-physi-hacker) [inline api/views.py:11]: this line is no longer needed, I guess?

File: api/views.py [hunk 10]
```diff
@@ -1,7 +1,28 @@
 from django.shortcuts import render
```
Comment (kali-physi-hacker) [inline api/views.py:2]: Also, you can take that off, we don't need that either?

**Review by kali-physi-hacker — CHANGES_REQUESTED** [review]

File: api/views.py [hunk 11]
```diff
@@ -1,7 +1,38 @@
...
     return render(request=request, template_name=template, context=context)
+
+
+@api_view(["GET"])
+def lyrics_list(request): # new 
+
+
+    if request.method == "GET":
+        
+        # get lyrics from model
```
[the remark this answers is not in the document]
Comment (RJ-0605) [reply api/views.py]: i think the current code is not spaced

**Review by kali-physi-hacker — APPROVED** [review]

**Pull request merged**